src/pages/navbar/PolicySearchPage.tsx

```diff
@@ -0,0 +1,125 @@
+import { PageContainer } from '@toolpad/core/PageContainer';
+import React, { useEffect, useState } from 'react';
+import { MeasurementResolution, PolicyDto } from '../../api/openAPI';
+import { usePolicyService } from '../../hooks/services/usePolicyService.ts';
+import {
+    Box,
+    Button,
+    CircularProgress,
+    DialogActions,
+    FormLabel,
+    Input,
+    NativeSelect,
+    Typography,
+} from '@mui/material';
+import SmartMeterPoliciesTable from '../../components/tables/SmartMeterPoliciesTable.tsx';
+import FormControl from '@mui/material/FormControl';
+
+const PolicySearchPage = () => {
+    const [policies, setPolicies] = useState<PolicyDto[] | undefined>(undefined);
+
+    const { searchPolicies } = usePolicyService();
+
+    useEffect(() => {
+        void loadPolicies();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const loadPolicies = async (maxPrice?: number, measurementResolution?: MeasurementResolution) => {
+        try {
+            const policies = await searchPolicies(maxPrice, measurementResolution);
+            setPolicies(policies);
+        } catch (error) {
+            console.error(error);
+        }
+    };
+
+    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        const data = new FormData(event.currentTarget);
+        let measurementResolution = data.get('measurementResolution') as MeasurementResolution | undefined;
+        let price = data.get('price') as unknown as number | undefined;
+
+        if (!measurementResolution) {
+            measurementResolution = undefined;
+        }
+
+        if (!price) {
+            price = undefined;
+        }
+
+        void loadPolicies(price, measurementResolution);
+    };
+
+    const handleReset = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
```

Comment:
By removing line 56 `event.preventDefault()` mentioned problem of no clearing the filter texts, would be solved.